File: web/src/lib/KindeAuthProvider.ts
import type {AuthenticationProvider} from '@microsoft/kiota-abstractions';
import type {RequestInformation} from "@microsoft/kiota-abstractions/dist/es/src/requestInformation";

export class KindeAuthProvider implements AuthenticationProvider {
    private readonly getToken: () => Promise<string | undefined>;

    constructor(getToken: () => Promise<string | undefined>) {
        this.getToken = getToken;
    }

    async authenticateRequest(request: RequestInformation): Promise<void> {
        const token = await this.getToken();
        if (token) {
            request.headers.add('Authorization', `Bearer ${token}`);
        }
    }
}

File: mobile/lib/api/kinde_auth_provider.dart
import 'package:microsoft_kiota_bundle/microsoft_kiota_bundle.dart';

import '../services/authentication_service.dart';

class KindeAuthenticationProvider implements AuthenticationProvider {
  final AuthenticationService authenticationService;

  KindeAuthenticationProvider(this.authenticationService);

  @override
  Future<void> authenticateRequest(
    RequestInformation request, [
    Map<String, Object>? additionalAuthenticationContext,
  ]) async {
    final token = await authenticationService.getToken();
    if (token != null) {
      request.headers["Authorization"] = {"Bearer $token"};
    }
  }
}
